### ISSUE TITLE
Inconsistent usage of this.database

### ISSUE THREAD

**Issue by JanneSalo** [a8m/doqmentdb#15]
I'm using doqmentdb 0.2.3. When you create a DoQmentDB object, you pass it the database name you want to use. The value is stored in `this.db`. Then, when you get a Collection object using `db.use('someCollection')`, the value of `this.db` is passed to Collection constructor. Nothing weird so far.
 
Everything is fine as long as `this.db` is the string I originally passed to DoQmentDB constructor. But in some cases, `this.db` in DoQmentDB is replaced by a database object and if I get new collection managers via `db.use()`, all hell breaks loose. The Collection constructor is now passed an object instead of a string, which is bad. The object is passed to `manager.findOrCreate()` (line 291 of `collection.js`) and all the way to `query()` in `lib/query.js`. When this happens, the "invalid operator" error gets thrown from line 46.

The error occurs at least when I search for collections using `db.find()` and wait for the promise to resolve (which I guess is also the point where `this.db` gets overridden internally) and then get a new collection object. A very simple piece of code to reproduce the issue:
```
	db.find().then(function(results) {
		var collection = db.use('some_collection');
		collection.find();
	});
```

**Comment by JanneSalo** [comment]
Hmm, could this maybe be fixed simply by adding a new property in the DoQmentDB object: `this.dbName`, initializing it in the constructor and then passing `this.dbName` to Collection constructor?

**Comment by a8m** [comment]
Thanks @JanneSalo 
Updated version **v0.2.4**, fix this bug and also added operations supporting, see: [README](https://github.com/a8m/doqmentdb#operations)
I'm just curious why you don't using `collection` simply like that ?
```js
var db = new DoQmentDB(...)
  , users = db.use('users');

users.find({})
  .then(console.log)
```

**Comment by JanneSalo** [comment]
Thanks! I just tested this and the issue indeed seems to be fixed.

There's a reason I'm getting the collections in such a peculiar way. It all comes down to working around DocumentDB limitations, really. I work on a project where the 10GB cap on collection might become an issue, so we've had to split some data that logically belongs in one collection into several collections. I wrote a little wrapper function to query across all collections at once, but in order to do that, I first need to know which collections exist in the database as simply getting collection manager for each possible collection (there aren't that many, but still) isn't an option because doing a `find()` on a collection that doesn't exist causes the collection to be created and that's where we run into another limitation of DocumentDB: only 3 collections per CU are allowed. That's why I need to know what to query before I query. :)
